common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -581,19 +577,19 @@ def handle_problem_html_error(self, err):
             self.set_score(self.score_from_lcp())
             # Prepend a scary warning to the student
             _ = self.runtime.service(self, "i18n").ugettext
-            warning_msg = _("Warning: The problem has been reset to its initial state!")
-            warning = '<div class="capa_reset"> <h2> ' + warning_msg + '</h2>'
+            warning_msg = Text(_("Warning: The problem has been reset to its initial state!"))
+            warning = HTML('<div class="capa_reset"> <h2>{}</h2>').format(warning_msg)
 
             # Translators: Following this message, there will be a bulleted list of items.
             warning_msg = _("The problem's state was corrupted by an invalid submission. The submission consisted of:")
-            warning += warning_msg + '<ul>'
+            warning += HTML('{}<ul>').format(warning_msg)
 
             for student_answer in student_answers.values():
                 if student_answer != '':
-                    warning += '<li>' + cgi.escape(student_answer) + '</li>'
+                    warning += HTML('<li>{}</li>').format(cgi.escape(student_answer))
```

Comment:
I think the `cgi.escape` calls (here and elsewhere) were just doing HTML escaping.  Now that you are using `HTML()`, this should now be as follows to avoid double-escaping:
```
warning += HTML('<li>{}</li>').format(student_answer)
```
I don't plan on commenting them all, so just do a search for `cgi.escape` in your PR and ensure they all are using `HTML` or `Text` as needed, and they have been removed.  Thanks.